scripts/build-bootstraps.sh

```diff
@@ -419,77 +604,90 @@ main() {
-		PACKAGES+=("gawk")
-		PACKAGES+=("grep")
-		PACKAGES+=("gzip")
-		PACKAGES+=("less")
-		PACKAGES+=("procps")
-		PACKAGES+=("psmisc")
-		PACKAGES+=("sed")
-		PACKAGES+=("tar")
-		PACKAGES+=("termux-core")
-		PACKAGES+=("termux-exec")
-		PACKAGES+=("termux-keyring")
-		PACKAGES+=("termux-tools")
-		PACKAGES+=("util-linux")
+		PACKAGES_LIST+=("coreutils")
+		PACKAGES_LIST+=("curl")
+		PACKAGES_LIST+=("dash")
+		PACKAGES_LIST+=("diffutils")
+		PACKAGES_LIST+=("findutils")
+		PACKAGES_LIST+=("gawk")
+		PACKAGES_LIST+=("grep")
+		PACKAGES_LIST+=("gzip")
+		PACKAGES_LIST+=("less")
+		PACKAGES_LIST+=("procps")
+		PACKAGES_LIST+=("psmisc")
+		PACKAGES_LIST+=("sed")
+		PACKAGES_LIST+=("tar")
+		PACKAGES_LIST+=("termux-core")
+		PACKAGES_LIST+=("termux-exec")
+		PACKAGES_LIST+=("termux-keyring")
+		PACKAGES_LIST+=("termux-tools")
+		PACKAGES_LIST+=("util-linux")
+		PACKAGES_LIST+=("xz-utils")
 
 		# Additional.
-		PACKAGES+=("ed")
-		PACKAGES+=("debianutils")
-		PACKAGES+=("dos2unix")
-		PACKAGES+=("inetutils")
-		PACKAGES+=("lsof")
-		PACKAGES+=("nano")
-		PACKAGES+=("net-tools")
-		PACKAGES+=("patch")
-		PACKAGES+=("unzip")
-
-		# Handle additional packages.
-		for add_pkg in "${ADDITIONAL_PACKAGES[@]}"; do
-			if [[ " ${PACKAGES[*]} " != *" $add_pkg "* ]]; then
-				PACKAGES+=("$add_pkg")
+		PACKAGES_LIST+=("ed")
+		PACKAGES_LIST+=("debianutils")
+		PACKAGES_LIST+=("dos2unix")
+		PACKAGES_LIST+=("inetutils")
+		PACKAGES_LIST+=("lsof")
+		PACKAGES_LIST+=("nano")
+		PACKAGES_LIST+=("net-tools")
+		PACKAGES_LIST+=("patch")
+		PACKAGES_LIST+=("unzip")
+
+		# Append `PACKAGES_POST_ADDITIONAL_LIST` to `PACKAGES_LIST`.
+		for package_name in "${PACKAGES_POST_ADDITIONAL_LIST[@]}"; do
+			if [[ " ${PACKAGES_LIST[*]} " != *" $package_name "* ]]; then
+				PACKAGES_LIST+=("$package_name")
 			fi
 		done
-		unset add_pkg
 
 		# Build packages.
-		for package_name in "${PACKAGES[@]}"; do
+		for package_name in "${PACKAGES_LIST[@]}"; do
 			set +e
-			build_package "$TERMUX_ARCH" "$package_name" || return $?
+			build_package "$package_arch" "$package_name" || return $?
 			set -e
 		done
 
-		# Extract all debs.
-		extract_debs "$TERMUX_ARCH" || return $?
+		# Extract all package files.
+		extract_package_files "$package_arch" "${PACKAGES_LIST[@]}"
 
 		# Add termux bootstrap second stage files
 		add_termux_bootstrap_second_stage_files "$package_arch"
 
 		# Create bootstrap archive.
-		create_bootstrap_archive "$TERMUX_ARCH" || return $?
+		create_bootstrap_archive "$package_arch"
 
+		elapsed_seconds=$(($(date +%s) - boostrap_build_start_time))
+		echo "[*] Building bootstrap for arch '$package_arch' completed in $((elapsed_seconds / 3600 )) hours $(((elapsed_seconds % 3600) / 60)) minutes $((elapsed_seconds % 60)) seconds"
 	done
 
+	elapsed_seconds=$(($(date +%s) - boostraps_build_start_time))
+	echo "[*] Building bootstraps completed in $((elapsed_seconds / 3600 )) hours $(((elapsed_seconds % 3600) / 60)) minutes $((elapsed_seconds % 60)) seconds"
+
```

Comment:
Probably we should move this code to a separate source file and reuse it here.
https://github.com/termux/termux-packages/blob/354126893c53a8fec95ae5cc761e1cc9a079d9cc/scripts/bin/update-packages#L85-L89